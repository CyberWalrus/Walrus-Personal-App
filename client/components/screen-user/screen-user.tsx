import { User } from "@client/type/data";
import { ComponentClass, PureComponent, ReactElement } from "react";
import * as React from "react";
import { connect } from "react-redux";
import { Operation } from "../../store/data/data";
import { getUser } from "../../store/data/selectors";
import { StateApp, ThunkDispatch } from "../../type/reducer";

interface PropsInsert {
  id: string;
}
interface PropsState {
  user: User;
}
interface PropsDispatch {
  onUserLoad: (id: string) => void;
}
type Props = PropsInsert & PropsState & PropsDispatch;

class ScreenUser extends PureComponent<Props> {
  public componentDidMount(): void {
    const id = this.props.id;
    this.props.onUserLoad(id);
  }
  public render(): ReactElement {
    const { user }: Props = this.props;
    return (
      <main className={`page-content`}>
        <section className={`form-custom-box`}>
          {user ? (
            <div>
              <div>{user.id}</div>
              <div>{user.login}</div>
            </div>
          ) : (
            <React.Fragment />
          )}
        </section>
      </main>
    );
  }
}

const mapStateToProps = (state: StateApp, ownProps: Props): Props => ({
  ...ownProps,
  user: getUser(state),
});

const mapDispatchToProps = (dispatch: ThunkDispatch): PropsDispatch => ({
  onUserLoad: (id: string): void => {
    dispatch(Operation.getUser(id));
  },
});

export { ScreenUser };

export default connect<Props, PropsDispatch, PropsInsert, StateApp>(
  mapStateToProps,
  mapDispatchToProps,
)(ScreenUser) as ComponentClass<PropsInsert>;
