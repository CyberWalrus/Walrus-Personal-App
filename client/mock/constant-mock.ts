import { FormType } from "../../config/constants";
import { Option } from "../hocs/with-form-state/with-form-state";
import { InputCustomInterface } from "../type/component";
import { User, UserRole } from "../type/data";

// tslint:disable-next-line:no-empty
const _functVoid = (): void => {};

const FORM_TYPE: FormType = FormType.SIGN_IN;

const user: User = {
  id: `test`,
  email: `test`,
  isActive: true,
  nickName: `test`,
  login: `test`,
  firstName: `test`,
  lastName: `test`,
  userRoleId: [`test`],
  signUpDate: 1234567,
  created: 1234567,
};
const userRole: UserRole = {
  id: `test`,
  name: `test`,
  isActive: true,
};
const optionForm: Option = {
  values: {},
  formErrors: {},
  formValid: true,
  onChangeUserInput: _functVoid,
  onClickSubmit: _functVoid,
};
const optionInput: InputCustomInterface = {
  formTypes: [FormType.SIGN_UP],
  type: `text`,
  name: `login`,
  placeHolder: `Login`,
  hidenValue: `Login`,
};

const usersEmpty: User[] = [];
const users: User[] = [user, user];

const userRolesEmpty: UserRole[] = [];
const userRoles: UserRole[] = [userRole, userRole];

export {
  user,
  usersEmpty,
  users,
  optionForm,
  FORM_TYPE,
  optionInput,
  userRole,
  userRolesEmpty,
  userRoles,
};
