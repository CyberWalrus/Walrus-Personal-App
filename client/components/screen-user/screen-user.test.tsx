import * as Enzyme from "enzyme";
import { shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson, { Json } from "enzyme-to-json";
import * as React from "react";
import { user } from "../../mock/constant-mock";
import { ScreenUser } from "./screen-user";

Enzyme.configure({ adapter: new Adapter() });
it(`ScreenUser correctly render without data`, (): void => {
  const jestFunc = jest.fn();
  const tree: Json = toJson(
    shallow(<ScreenUser onUserLoad={jestFunc} user={undefined} id={`test`} />),
  );
  expect(tree).toMatchSnapshot();
});

it(`ScreenUser correctly render without data`, (): void => {
  const jestFunc = jest.fn();
  const tree: Json = toJson(
    shallow(<ScreenUser onUserLoad={jestFunc} user={user} id={user.id} />),
  );
  expect(tree).toMatchSnapshot();
});
