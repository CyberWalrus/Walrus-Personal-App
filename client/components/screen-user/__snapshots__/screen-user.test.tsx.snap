// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenUser correctly render without data 1`] = `
<main
  className="page-content"
>
  <section
    className="form-custom-box"
  />
</main>
`;

exports[`ScreenUser correctly render without data 2`] = `
<main
  className="page-content"
>
  <section
    className="form-custom-box"
  >
    <div>
      <div>
        test
      </div>
      <div>
        test
      </div>
    </div>
  </section>
</main>
`;
